lib/pluginmanager/update.rb

```diff
@@ -62,6 +63,20 @@ def update_gems!
     display_bundler_output(output)
   end
 
+  # validate if there is any major version update so then we can ask the user if he is
+  # sure to update or not.
+  def validate_major_version(plugin)
+    require "gems"
+    latest_version  = Gems.versions(plugin)[0]['number'].split(".")
```

Comment:
@talevy as I understand the code base nowadays, all plugins installed as local gems are skip, so this function will only get executed for plugins that are present in rubygems because they got installed throw them.

In detail, this https://github.com/elastic/logstash/blob/master/lib/pluginmanager/update.rb#L44 line reject plugins that are installed with https://github.com/elastic/logstash/blob/master/lib/pluginmanager/update.rb#L6, so all local fields. This is why we see this message:

```
╰─$ bin/plugin update logstash-input-log4j
Update is not supported for manually defined plugins or local .gem plugin installations, skipping: logstash-input-log4j
No plugin updated
```
if we try to update with a local gem installed. And for now there is no option to update a gem installed from a rubygems with a gem locally as all calls goes throw our beloved Bundler.

Makes sense to you @talevy ?